staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -195,10 +195,13 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 		}
 
 		config, err := LoadFromFile(filename)
+
 		if os.IsNotExist(err) {
 			// skip missing files
+			errlist = append(errlist, fmt.Errorf("No such file or directory %s", filename))
```

Comment:
explicitly referenced files are already mentioned in errors on line 180. Erroring here on implicit files (default path or $KUBECONFIG values) breaks configurations that currently work

Reply:
thank you liggitt for the more explanation. :smiley:

Reply:
@prksu @liggitt Thanks for reviewing it. And yeah it is breaking the existing flow. 
If we can't error out the bad files ( or not present files) here does that mean we, we won't be able to change the error message?  

The explicitly referenced files errors don't error out when the file is not present. L180

Reply:
> The explicitly referenced files errors don't error out when the file is not present. L180

Files referenced via flag do error out.

I think the best we can do for missing files referenced via envvar is to warn if all the files are missing.

Something like this:
```diff
diff --git a/staging/src/k8s.io/client-go/tools/clientcmd/loader.go b/staging/src/k8s.io/client-go/tools/clientcmd/loader.go
index e00ea38272d..751f9db5ddc 100644
--- a/staging/src/k8s.io/client-go/tools/clientcmd/loader.go
+++ b/staging/src/k8s.io/client-go/tools/clientcmd/loader.go
@@ -124,6 +124,8 @@ type ClientConfigLoadingRules struct {
        // that a default object that doesn't set this will usually get the behavior it wants.
        DoNotResolvePaths bool
 
+       WarnIfAllMissing bool
+
        // DefaultClientConfig is an optional field indicating what rules to use to calculate a default configuration.
        // This should match the overrides passed in to ClientConfig loader.
        DefaultClientConfig ClientConfig
@@ -137,19 +139,22 @@ var _ ClientConfigLoader = &ClientConfigLoadingRules{}
 func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
        chain := []string{}
 
+       warnIfAllMissing := false
        envVarFiles := os.Getenv(RecommendedConfigPathEnvVar)
        if len(envVarFiles) != 0 {
                fileList := filepath.SplitList(envVarFiles)
                // prevent the same path load multiple times
                chain = append(chain, deduplicate(fileList)...)
+               warnIfAllMissing = true
 
        } else {
                chain = append(chain, RecommendedHomeFile)
        }
 
        return &ClientConfigLoadingRules{
-               Precedence:     chain,
-               MigrationRules: currentMigrationRules(),
+               WarnIfAllMissing: warnIfAllMissing,
+               Precedence:       chain,
+               MigrationRules:   currentMigrationRules(),
        }
 }
 
@@ -172,6 +177,7 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
        }
 
        errlist := []error{}
+       missing := []string{}
 
        kubeConfigFiles := []string{}
 
@@ -197,6 +203,7 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
                config, err := LoadFromFile(filename)
                if os.IsNotExist(err) {
                        // skip missing files
+                       missing = append(missing, filename)
                        continue
                }
                if err != nil {
@@ -207,6 +214,10 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
                kubeconfigs = append(kubeconfigs, config)
        }
 
+       if rules.WarnIfAllMissing && len(missing) > 0 && len(kubeconfigs) == 0 {
+               klog.Warningf("Config not found: %s", strings.Join(missing, ", "))
+       }
+
        // first merge all of our maps
        mapConfig := clientcmdapi.NewConfig()
```

Reply:
It's better this way, to warn on the missing files. Made modifications as suggested.
 Thanks @liggitt!